tests/test_generate_index.py

```diff
@@ -191,8 +192,8 @@ def test_abort_generate_release_index_source_branch_not_exists(setup_index_tests
         generate_release_index(target_branch, Path(target_worktree), not_source_branch, target_branch_exists_remote = True)
 
     assert e.match(f".* not currently checked out. Please Check out branch .*"
-                   f" before calling generate_release_index.")
-    assert e.type == SystemExit
+                   f" before calling generate_release_index.") \
+           and e.type == SystemExit
```

Comment:
You may also wan't to consider making it a `multi line assert` using the `()` syntax.